pyhap/accessory_driver.py

```diff
@@ -185,6 +159,9 @@ def __init__(self, accessory, port, address=None, persist_file="accessory.state"
             self.persist()
         self.topics = {}  # topic: set of (address, port) of subscribed clients
         self.topic_lock = threading.Lock()  # for exclusive access to the topics
+        self.event_loop = asyncio.get_event_loop()
+        self.aio_stop_event = asyncio.Event()
```

Comment:
I think there is a better, cleaner way to do this.

It should be possible to call `self.stop_event` using `event_loop.run_in_executor()` to wait on the threaded event in a separate thread managed by asyncio. Then put a callback on the returned future to call `aio_stop_event.set()` back in the main thread.

That way calling `stop_event.set()` anywhere is both threadsafe, and guaranteed to both threads and the `event_loop`.

I'll put together a PR once I figured out the nicest way the structure the code.

Reply:
I agree that it will be better. Do you mind if we leave it for another change?